File: backend/Models/Verse.cs
namespace Encourager.Api.Models;

public record Verse(string Text, string Reference);

public record VerseMetadata(int VerseId, string Book, int Chapter, string VerseNumber);

public record VerseText(string Text, string Language);

public record VerseResponse(
    int VerseId,
    string Book,
    int Chapter,
    string VerseNumber,
    string Text,
    string Language,
    IReadOnlyDictionary<string, string> Translations);

File: frontend/src/types/verse.ts
export interface Verse {
  verseId: number
  book: string
  chapter: number
  verseNumber: string
  text: string
  language: string
  translations: Record<string, string>
}

export interface BlessingData {
  timestamp: string
  verse: Verse
}

export interface LegacyVerse {
  text: string
  reference: string
  index: number
}

export interface LegacyBlessingData {
  timestamp: string
  verse: LegacyVerse
}
